### ISSUE TITLE
Charts not displaying when using ChangeDetectionStrategy.OnPush

### ISSUE THREAD

**Issue by DavidMarquezF** [swimlane/ngx-charts#917]
**I'm submitting a ...**  (check one with "x")

- [x ] bug report - search github for a similar issue or PR before submitting
- [ ] support request - use StackOverflow (add the `ngx-charts` tag) or the gitter chat for support questions
- [ ] feature request

**Current behavior**
I'm developping an editable dashboard that loads dynamically its widgets. I use ngx-charts in a lot of them and when I changed the ChangeDetectionStrategy to OnPush for a parent component, the charts do not draw. 

However, when I hover my mouse over where the tooltips area would be if the charts were drawn, all the charts of that dashboard draw themselves. I think this may be caused because the tooltip area has an output that triggers Change Detection (I think it was discussed in this issue #439 ).

I really need them to work with OnPush because changeDetection was impacting too much my performance.

**Expected behavior**
The charts should be drawn regardless of the ChangeDetectionStrategy method.

**Reproduction of the problem**
![image](https://user-images.githubusercontent.com/32216320/42800936-026b1c00-899e-11e8-9533-386e903ac5a7.png)
 As you can see. The dimensions are what they should be but the charts are not drawn.

However when I hover the mouse over where the tooltip are would be if they were drawn:
![image](https://user-images.githubusercontent.com/32216320/42801024-5003a446-899e-11e8-881a-64e78744bd9b.png)

**What is the motivation / use case for changing the behavior?**
Improving performance and "support" ChangeDetectionStrategy feature

**Please tell us about your environment:**
Windows 10, Visual Studio, npm

* **ngx-charts version:** 8.1.0

* **Angular version:** 6.2.1

* **Browser:** 
All

* **Language:** TypeScript 2.7.2

**Comment by marjan-georgiev** [comment]
Interesting. All of the charts themselves are using `ChangeDetectionStrategy.OnPush`.

Here's a stackblitz demo where the parent component is using OnPush and it seems to be working as expected: https://stackblitz.com/edit/vertical-bar-chart-3zepui?file=app/app.component.ts

Could you create a stackblitz where what you see can be reproduced please?

**Comment by DavidMarquezF** [comment]
You're right. I reproduced exactly what I have in my project in this stackblitz: [https://stackblitz.com/edit/vertical-bar-chart-4txsnh](url) and it worked lile a charm. 

I investigated some more and remembered that I used a directive that detaches and re-attaches a given element if it's in the DOM. This helped when charts didn't resize properly or animations that didn't wait to route to be loaded completely.

Once I removed it, it worked (guess I'll have to fix it/find another way to do it).

Either way, thank you and sorry for the inconvenience
